lightningd/plugin.c

```diff
@@ -900,6 +900,21 @@ void clear_plugins(struct plugins *plugins)
 		tal_free(p);
 }
 
+void plugins_add_default_dir(struct plugins *plugins, const char *default_dir)
+{
+	DIR *d = opendir(default_dir);
+	if (d) {
+		struct dirent *di;
+		while ((di = readdir(d)) != NULL) {
+			if (streq(di->d_name, ".") || streq(di->d_name, ".."))
+				continue;
+			add_plugin_dir(plugins, path_join(NULL, default_dir, di->d_name), true);
+		}
+	}
+	closedir(d);
+    tal_free(default_dir);
```

Comment:
Indent.  Also, it's an anti-pattern to take ownership of a parameter's lifetime like this without explicit indication to the caller.  I would just use tmpctx instead of NULL here, and in the caller.

Reply:
I added that `tal_free` in a hurry to make travis pass, I wasn't aware of : 
```
* allocate from.  We have a magic convenience context
* `tmpctx` for temporary allocations like this.
```
it's useful ^^.